frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1642,6 +1723,9 @@ mod feasibility_check {
 				MultiPhase::feasibility_check(solution, COMPUTE),
 				FeasibilityError::SnapshotUnavailable
 			);
+
+			// restore noop snapshot to ensure post-test `try_state` checks pass.
+			<Snapshot<Runtime>>::set(Some(Default::default()));
```

Comment:
or may be kill all three `snapshot`, `snapshot_metadata` and `desired_targets` to ensure storage is consistent? Would be a good documentation of the expected behaviour that either all exists or none for future readers.

Reply:
I agree, I changed the code to kill both `desired_targets` and `snapshot_metadata`.

There are 2 places where the invariants related to these storage types are defined:

1. lib.rs storage defs.

```
    /// Snapshot data of the round.
	///
	/// This is created at the beginning of the signed phase and cleared upon calling `elect`.
    /// ...

	/// Desired number of targets to elect for this round.
	///
	/// Only exists when [`Snapshot`] is present.
    /// ...

	/// The metadata of the [`RoundSnapshot`]
	///
	/// Only exists when [`Snapshot`] is present.
    /// ...
```

2. try-state checks.